src/pages/Proposal/components/Vote/VoteAction.tsx

```diff
@@ -0,0 +1,189 @@
+import React, { FC, ReactNode } from 'react';
+import { Link } from 'react-router-dom';
+
+import { isNil } from 'lodash';
+import type { Choice, Proposal } from '@zero-tech/zdao-sdk';
+import { ProposalState } from '@zero-tech/zdao-sdk';
+import {
+	useCurrentDao,
+	useWeb3,
+	useUserProposalVoteData
+} from '../../../../lib/hooks';
+import {
+	isFromSnapshotWithMultipleChoices,
+	getSnapshotProposalLink
+} from '../../../../features/view-dao-proposals/DaoProposalsTable/lib';
+
+import { LoadingIndicator, Tooltip } from '@zero-tech/zui/components';
+import { ConnectWallet } from '../../../../features/ui';
+import { VoteButtons, Approve, Deny } from './VoteButtons';
+import { VoteStatus } from './Vote';
+
+import externalIcon from '../../../../assets/external-link.svg';
+
+import classNames from 'classnames';
+import styles from './Vote.module.scss';
+
+////////////////
+// VoteAction //
+////////////////
+
+interface VoteActionProps {
+	proposal: Proposal;
+	userChoice: Choice;
+	voteStatus: VoteStatus;
+	onClickApprove: () => void;
+	onClickDeny: () => void;
+}
+
+/**
+ * Renders actions based on props.
+ * @param proposal Proposal
+ * @param userChoice the vote the user made, if any
+ * @param voteStatus current progress of voting modal
+ * @param onClickApprove event fired when approve is clicked
+ * @param onClickDeny event fired when deny is clicked
+ */
+export const VoteAction: FC<VoteActionProps> = ({
+	proposal,
+	userChoice,
+	voteStatus,
+	onClickApprove,
+	onClickDeny
+}) => {
+	const { dao, zna } = useCurrentDao();
+	const { account } = useWeb3();
+	const { isLoading, data: { userVote, userVotingPower } = {} } =
+		useUserProposalVoteData({ zna, proposalId });
```

Comment:
Hey sorry mate one other change needed:

<img width="1437" alt="Screenshot 2023-04-04 at 14 30 09" src="https://user-images.githubusercontent.com/39112648/229687386-5b1ba87c-f9a7-412b-bf39-12515cdc64c1.png">

I spotted this when removing `dao.listAssetsCollectibles();` from the `useDaoAssets` hook which meant the proposals would load and I could open/navigate to them.

```
const { dao, zna } = useCurrentDao();
const { account } = useWeb3();
const proposalId = proposal?.id;
const { isLoading, data: { userVote, userVotingPower } = {} } =
		useUserProposalVoteData({ zna, proposalId });
```

Reply:
`useUserProposalVoteData({ zna, proposalId: proposal?.id }); `

Reply:
Thank u, have fixed!